integration/src/integration/java/pl/allegro/tech/hermes/integration/JmsConsumingTest.java

```diff
@@ -64,6 +65,43 @@ public void shouldPublishAndConsumeJMSMessageWithTraceId() throws Exception {
         assertThat(jmsEndpoint.waitAndGetLastMessage()).assertStringProperty("TraceId", traceId);
     }
 
+    @Test
+    public void shouldPublishAndConsumeJmsMessageWithTraceHeaders() throws Exception {
```

Comment:
This test could be a little more concise by using instance of ```Invocation.Builder request``` object in ```// given``` block and then in ```// when``` it could be just posted in one line. Then it would be possible to just iterate over all the headers from request and compare them with response if they are the same.